src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -273,13 +291,21 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm)
 
             var booleanQuery = new BooleanQuery();
             string searchTerm = QueryParser.Escape(SearchTerm);
-            var bCategoryBasedSearch = searchTerm.Contains('.') ? true : false;
+
+            if (searchTerm.Contains('.'))
+                searchType = SearchType.ByCategory;
+            else if (searchTerm.Contains(' '))
+                searchType = SearchType.ByEmptySpace;
+            else
+                searchType = SearchType.Normal;
+
+            var trimmedSearchTerm = searchType == SearchType.ByEmptySpace ? searchTerm.Replace(" ", "") : searchTerm;
```

Comment:
What if the actual node name has spaces in it? Will those show up with this condition?
Also since the ByEmptySpace is just used to check if the search term has to be trimmed or not, I don't think it should be a value inside SearchType enum.

Reply:
@reddyashish for the first question I think the Tokenizer that we are using is removing the empty spaces, just to confirm I've tested in **DynamoSandbox** with several nodes that use empty spaces and I'm getting the expected node always in first place (see image below) so there should not be any problem with nodes having empty space.
![image](https://github.com/DynamoDS/Dynamo/assets/61755417/74618367-af34-4e13-8714-eb2399f143e5)

For the second comment we are using SearchType.ByEmptySpace in more conditions in the code that's why I considered to add it in the SearchType enum.
![image](https://github.com/DynamoDS/Dynamo/assets/61755417/abb00bbd-d11f-46ff-b997-1a96722e3251)

Reply:
Thank you for the confirmation @RobertGlobant20. Can't we just use that as a property to check for empty spaces then? Because the search algorithm is not altered, just the search term. Might not be a big thing.

Reply:
Converted to be a property in the next commit: https://github.com/DynamoDS/Dynamo/pull/14878/commits/ff1b100724613ce899caaa4ff62097b28f0bdfaf